tests/src/whisk/core/apigw/actions/test/ApiGwRoutemgmtActionTests.scala

```diff
@@ -0,0 +1,470 @@
+    val config = new WhiskConfig(WhiskAuthStore.requiredProperties ++ WhiskEntityStore.requiredProperties)
+    val authstore = WhiskAuthStore.datastore(config)
+    val whiskkey = Await.result(WhiskAuth.get(authstore, Subject("whisk.system"),false)(TransactionId.unknown), 5.seconds)
+    val whisksystemAuthId = whiskkey.authkey.uuid.toString
+    val whisksystemAuthPwd = whiskkey.authkey.key.toString
+
+    override def afterAll() {
+        println("Shutting down authstore connection")
+        authstore.shutdown()
+        super.afterAll()
+    }
+
+    def getApis(
+               bpOrName: String,
+               relpath: String = "",
+               operation: String = "",
+               docid: String = ""): JsArray = {
+        val parms = Map[String,JsValue]() ++
+            Map("namespace" -> "_".toJson) ++
+            { if (bpOrName.length() > 0)  Map("basepath" -> bpOrName.toJson) else Map[String,JsValue]() } ++
+            { if (relpath.length() > 0)   Map("relpath" -> relpath.toJson) else Map[String,JsValue]() } ++
+            { if (operation.length() > 0) Map("operation" -> operation.toJson) else Map[String,JsValue]() } ++
+            { if (docid.length() > 0)     Map("docid" -> docid.toJson) else Map[String,JsValue]() }
+        val wskprops = new WskProps(authKey = whisksystemAuthId+":"+whisksystemAuthPwd)
+        val rr = wsk.action.invoke(
+            name = "routemgmt/getApi",
+            parameters = parms,
+            blocking = true,
+            result = true,
+            expectedExitCode = SUCCESS_EXIT)(wskprops)
+        var apiJsArray: JsArray =
+            try {
+                var apisobj = rr.stdout.parseJson.asJsObject.fields("apis")
+                apisobj.convertTo[JsArray]
+            } catch {
+                case e:Exception =>
+                    JsArray.empty
+            }
+        return apiJsArray
+    }
+
+    def createApi(
+               namespace: String,
+               basepath: String = "/",
+               relpath: String,
+               operation: String,
+               apiname: String = "",
+               action: ApiAction = null,
+               swagger: String = "",
+               expectedExitCode: Int = SUCCESS_EXIT): RunResult ={
+        val parms = Map[String,JsValue]() ++
+            Map("namespace" -> "_".toJson) ++
+            { if (basepath.length() > 0)  Map("basepath" -> basepath.toJson) else Map[String,JsValue]() } ++
+            { if (relpath.length() > 0)   Map("relpath" -> relpath.toJson) else Map[String,JsValue]() } ++
+            { if (operation.length() > 0) Map("operation" -> operation.toJson) else Map[String,JsValue]() } ++
+            { if (apiname.length() > 0) Map("apiname" -> apiname.toJson) else Map[String,JsValue]() } ++
+            { if (action != null) Map("action" -> action.toJson) else Map[String,JsValue]() } ++
+            { if (swagger.length() > 0) Map("swagger" -> swagger.toJson) else Map[String,JsValue]() }
```

Comment:
Seems like you want all of those to be `Option[String]` instead of checking the length of the String?